File: rick.fs
open System
open System.Diagnostics

let openRickroll() =
    let url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
    if Environment.OSVersion.Platform = PlatformID.Win32NT then
        Process.Start("cmd", sprintf "/C start %s" url) |> ignore
    elif System.Runtime.InteropServices.RuntimeInformation.OSDescription.Contains("Darwin") then
        Process.Start("open", url) |> ignore
    else
        Process.Start("xdg-open", url) |> ignore // Assume Linux

[<EntryPoint>]
let main argv =
    let processInfo = ProcessStartInfo("curl", "ascii.live/rick")
    processInfo.RedirectStandardOutput <- true
    processInfo.RedirectStandardError <- true
    processInfo.UseShellExecute <- false

    use process = Process.Start(processInfo)
    let output = process.StandardOutput.ReadToEnd()
    let errorOutput = process.StandardError.ReadToEnd()
    
    process.WaitForExit()

    if process.ExitCode = 0 then
        printfn "%s" output // Print command output
    else
        printfn "Command failed: %s" errorOutput
        openRickroll()

    0 // Return an integer exit code

File: rick.go
package main

import (
    "fmt"
    "os/exec"
    "runtime"
    "strings"
)

func openBrowser(url string) {
    var cmd string
    if strings.Contains(runtime.GOOS, "windows") {
        cmd = "start"
    } else if runtime.GOOS == "darwin" { // macOS
        cmd = "open"
    } else { // Assume Linux
        cmd = "xdg-open"
    }
    exec.Command(cmd, url).Start()
}

func main() {
    cmd := exec.Command("curl", "ascii.live/rick")
    output, err := cmd.CombinedOutput()

    if err != nil {
        fmt.Println("Command failed:", err)
        openBrowser("https://www.youtube.com/watch?v=dQw4w9WgXcQ")
        return
    }
    fmt.Println(string(output)) // Print command output
}